Logstash migration guide (#1487)
opensearch-project/documentation-website: _clients/data-prepper/configure-logstash-data-prepper.md

---
layout: default
title: Configure Logstash for Data Prepper
parent: Data Prepper
nav_order: 2
---
# Configure Logstash for Data Prepper
You can run Data Prepper with a Logstash configuration.

> As mentioned in the [Getting Started]({{site.url}}{{site.baseurl}}/opensearch/clients/data-prepper/getting-started) guide, you'll need to configure Data Prepper with a pipeline using a `pipelines.yaml` file.

Reviewer: Getting Started guide --> Getting Started section?

> Alternatively, if you have a Logstash configuration `logstash.conf` to configure Data Prepper instead of `pipelines.yaml`.

Reviewer 2: It looks like there are some missing words here. Should it be "Alternatively, if you have a Logstash configuration, you can use..."?

Reviewer: I don't see this change?

> Alternatively, if you have a Logstash configuration `logstash.conf` to configure Data Prepper instead of `pipelines.yaml`.

Author: ```suggestion
Alternatively, if you have a Logstash configuration, you can use `logstash.conf` to configure Data Prepper instead of `pipelines.yaml`.
```

Author: @natebower Incorporated feedback so it includes "you can use" now.

> ## Supported Plugins

Reviewer 2: ```suggestion
## Supported plugins
```

As of the Data Prepper 1.2 release, the following plugins from the Logstash configuration are supported:
* HTTP Input plugin
* Grok Filter plugin
* Elasticsearch Output plugin
* Amazon Elasticsearch Output plugin

## Limitations

> * Apart from the supported plugins, all other plugins from the Logstash configuration will throw an `Exception` and fail to execute.

Reviewer 2: ```suggestion
* Apart from the supported plugins, all other plugins from the Logstash configuration will throw an `Exception` and fail to run.
```

> * Conditionals in the Logstash configuration are not supported as of Data Prepper 1.2 release.

Reviewer 2: ```suggestion
* Conditionals in the Logstash configuration are not supported as of the Data Prepper 1.2 release.
```

> ## Running Data Prepper with Logstash Configuration

Reviewer 2: ```suggestion
## Running Data Prepper with a Logstash configuration
```

> ## Running Data Prepper with a Logstash configuration

Reviewer: Can we have an intro paragraph here? Like To run Data Prepper with a Logstash configuration, do the following.

> 1. To install Data Prepper's docker image, visit the _Installation_ section in [Getting Started]({{site.url}}{{site.baseurl}}/opensearch/clients/data-prepper/getting_started).

Reviewer 2: ```suggestion
1. To install Data Prepper's Docker image, see the _Installation_ section in the [Getting Started]({{site.url}}{{site.baseurl}}/opensearch/clients/data-prepper/getting_started) guide.
```

> 1. To install Data Prepper's docker image, visit the _Installation_ section in [Getting Started]({{site.url}}{{site.baseurl}}/opensearch/clients/data-prepper/getting_started).

Reviewer 2: Can this be "To install the Data Prepper Docker image" in order to avoid using punctuation with "Data Prepper"?

> 1. To install Data Prepper's Docker image, see the _Installation_ section in the [Getting Started]({{site.url}}{{site.baseurl}}/opensearch/clients/data-prepper/getting_started) guide.

Reviewer: I think we need a command here, like "Install the Data Prepper Docker image. For instructions, see .... 

Remove "guide".

> 2. Run the docker image pulled in Step 1 by supplying your `logstash.conf` configuration.

Reviewer 2: ```suggestion
2. Run the Docker image installed in Step 1 by supplying your `logstash.conf` configuration.
```

```
docker run --name data-prepper -p 4900:4900 -v ${PWD}/logstash.conf:/usr/share/data-prepper/pipelines.conf opensearchproject/data-prepper:latest pipelines.conf
```

> The `logstash.conf` file gets converted to `logstash.yaml` by mapping the plugins and attributes in the Logstash configuration to the corresponding plugins and attributes in Data Prepper.

Reviewer 2: ```suggestion
The `logstash.conf` file is converted to `logstash.yaml` by mapping the plugins and attributes in the Logstash configuration to the corresponding plugins and attributes in Data Prepper.
```

> You can find the converted `logstash.yaml` file in the same directory where you placed `logstash.conf`.

Reviewer 2: Is there a more precise/technical term than "placed"?

Author: Replaced "placed" with "stored".

> You can find the converted `logstash.yaml` file in the same directory where you placed `logstash.conf`.

Author: ```suggestion
You can find the converted `logstash.yaml` file in the same directory where you stored `logstash.conf`.
```
